### PR TITLE
executor: Remove unnecessary information in explain analyze output

### LINKED ISSUE #16224: Explain analyze output has redundant information.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?

```sql
use tpch;
explain analyze select
                          l_orderkey,
                          sum(l_extendedprice * (1 - l_discount)) as revenue,
                          o_orderdate,
                          o_shippriority
                  from
                          customer,
                          orders,
                          lineitem
                  where
                          c_mktsegment = 'AUTOMOBILE'
                          and c_custkey = o_custkey
                          and l_orderkey = o_orderkey
                          and o_orderdate < '1995-03-13'
                          and l_shipdate > '1995-03-13'
                  group by
                          l_orderkey,
                          o_orderdate,
                          o_shippriority
                  order by
                          revenue desc,
                          o_orderdate      limit 10;
+----------------------------------------+----------+---------+-----------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------+------+
| id                                     | estRows  | actRows | task      | access object                                           | execution info                                                                                                                                                                                                                                                            | operator info                                                                                                                                                                                                                                   | memory                | disk |
+----------------------------------------+----------+---------+-----------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------+------+
| Projection_14                          | 10.00    | 10      | root      |                                                         | time:2.774196411s, loops:2, Concurrency:OFF                                                                                                                                                                                                                               | tpch.lineitem.l_orderkey, Column#35, tpch.orders.o_orderdate, tpch.orders.o_shippriority                                                                                                                                                        | 2.484375 KB           | N/A  |
| └─TopN_17                              | 10.00    | 10      | root      |                                                         | time:2.774191962s, loops:2                                                                                                                                                                                                                                                | Column#35:desc, tpch.orders.o_orderdate:asc, offset:0, count:10                                                                                                                                                                                 | 76.71875 KB           | N/A  |
|   └─HashAgg_23                         | 15.62    | 11479   | root      |                                                         | time:2.772270046s, loops:13, PartialConcurrency:4, FinalConcurrency:4                                                                                                                                                                                                     | group by:Column#48, Column#49, Column#50, funcs:sum(Column#44)->Column#35, funcs:firstrow(Column#45)->tpch.orders.o_orderdate, funcs:firstrow(Column#46)->tpch.orders.o_shippriority, funcs:firstrow(Column#47)->tpch.lineitem.l_orderkey       | 769.875 KB            | N/A  |
|     └─Projection_81                    | 15.62    | 30266   | root      |                                                         | time:2.765709221s, loops:40, Concurrency:OFF                                                                                                                                                                                                                              | mul(tpch.lineitem.l_extendedprice, minus(1, tpch.lineitem.l_discount))->Column#44, tpch.orders.o_orderdate, tpch.orders.o_shippriority, tpch.lineitem.l_orderkey, tpch.lineitem.l_orderkey, tpch.orders.o_orderdate, tpch.orders.o_shippriority | 641.484375 KB         | N/A  |
|       └─IndexMergeJoin_36              | 15.62    | 30266   | root      |                                                         | time:2.759902421s, loops:40, Concurrency:4                                                                                                                                                                                                                                | inner join, inner:Projection_34, outer key:tpch.orders.o_orderkey, inner key:tpch.lineitem.l_orderkey                                                                                                                                           | 4.172118186950684 MB  | N/A  |
|         ├─IndexMergeJoin_67(Build)     | 12.50    | 144399  | root      |                                                         | time:2.159646253s, loops:161, Concurrency:4                                                                                                                                                                                                                               | inner join, inner:TableReader_65, outer key:tpch.orders.o_custkey, inner key:tpch.customer.c_custkey                                                                                                                                            | 27.962088584899902 MB | N/A  |
|         │ ├─TableReader_74(Build)      | 3323.33  | 726007  | root      |                                                         | time:648.96198ms, loops:709, rpc num: 3, rpc max:774.231387ms, min:364.233446ms, avg:622.011455ms, p80:774.231387ms, p95:774.231387ms, proc keys max:583685, p95:583685                                                                                                   | data:Selection_73                                                                                                                                                                                                                               | 17.273679733276367 MB | N/A  |
|         │ │ └─Selection_73             | 3323.33  | 726007  | cop[tikv] |                                                         | proc max:710ms, min:328ms, p80:710ms, p95:710ms, iters:1479, tasks:3                                                                                                                                                                                                      | lt(tpch.orders.o_orderdate, 1995-03-13 00:00:00.000000)                                                                                                                                                                                         | N/A                   | N/A  |
|         │ │   └─TableFullScan_72       | 10000.00 | 1500000 | cop[tikv] | table:orders                                            | proc max:671ms, min:305ms, p80:671ms, p95:671ms, iters:1479, tasks:3                                                                                                                                                                                                      | keep order:false, stats:pseudo                                                                                                                                                                                                                  | N/A                   | N/A  |
|         │ └─TableReader_65(Probe)      | 0.00     | 126276  | root      |                                                         | time:5.12548383s, loops:182, rpc num: 36, rpc max:194.372702ms, min:19.944471ms, avg:139.670737ms, p80:180.743329ms, p95:193.963181ms, proc keys max:22451, p95:22430                                                                                                     | data:Selection_64                                                                                                                                                                                                                               | N/A                   | N/A  |
|         │   └─Selection_64             | 0.00     | 126276  | cop[tikv] |                                                         | proc max:181ms, min:9ms, p80:159ms, p95:176ms, iters:776, tasks:36                                                                                                                                                                                                        | eq(tpch.customer.c_mktsegment, "AUTOMOBILE")                                                                                                                                                                                                    | N/A                   | N/A  |
|         │     └─TableRangeScan_63      | 1.00     | 635411  | cop[tikv] | table:customer                                          | proc max:176ms, min:9ms, p80:155ms, p95:171ms, iters:776, tasks:36                                                                                                                                                                                                        | range: decided by [tpch.orders.o_custkey], keep order:true, stats:pseudo                                                                                                                                                                        | N/A                   | N/A  |
|         └─Projection_34(Probe)         | 1.25     | 30266   | root      |                                                         | time:3.940728772s, loops:53, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF, Concurrency:OFF | tpch.lineitem.l_orderkey, tpch.lineitem.l_extendedprice, tpch.lineitem.l_discount, tpch.lineitem.l_shipdate                                                                                                                                     | 6.6484375 KB          | N/A  |
|           └─IndexLookUp_33             | 1.25     | 30266   | root      |                                                         | time:3.940527604s, loops:53, rpc num: 21, rpc max:501.484633ms, min:6.379532ms, avg:163.101573ms, p80:282.034808ms, p95:347.375338ms, proc keys max:77027, p95:67291                                                                                                      |                                                                                                                                                                                                                                                 | 28.37890625 KB        | N/A  |
|             ├─IndexRangeScan_30(Build) | 3.75     | 577304  | cop[tikv] | table:lineitem, index:PRIMARY(L_ORDERKEY, L_LINENUMBER) | proc max:486ms, min:5ms, p80:270ms, p95:331ms, iters:657, tasks:21                                                                                                                                                                                                        | range: decided by [eq(tpch.lineitem.l_orderkey, tpch.orders.o_orderkey)], keep order:true, stats:pseudo                                                                                                                                         | N/A                   | N/A  |
|             └─Selection_32(Probe)      | 1.25     | 30266   | cop[tikv] |                                                         | proc max:193ms, min:1ms, p80:97ms, p95:171ms, iters:976, tasks:89                                                                                                                                                                                                         | gt(tpch.lineitem.l_shipdate, 1995-03-13 00:00:00.000000)                                                                                                                                                                                        | N/A                   | N/A  |
|               └─TableRowIDScan_31      | 3.75     | 577304  | cop[tikv] | table:lineitem                                          | proc max:191ms, min:1ms, p80:96ms, p95:168ms, iters:976, tasks:89                                                                                                                                                                                                         | keep order:false, stats:pseudo                                                                                                                                                                                                                  | N/A                   | N/A  |
+----------------------------------------+----------+---------+-----------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------+------+
```

As you can see, `Projection_34 ` has many duplicate `Concurrency:OFF` information.

<!-- If possible, provide a recipe for reproducing the error. -->

### 2. What did you expect to see?

No duplicate `Concurrency:OFF` information.

### 3. What did you see instead?

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)

```sql
mysql>select tidb_version();
+-------------------------------------------------------------------+
| tidb_version()                                                    |
+-------------------------------------------------------------------+
| Release Version: v4.0.0-beta.2-196-g6a45a7d6e                     |
| Git Commit Hash: 6a45a7d6eee51cc67e42f510b203eeb256da634d         |
| Git Branch: master                                                |
| UTC Build Time: 2020-04-08 08:14:25                               |
| GoVersion: go1.13.4                                               |
| Race Enabled: false                                               |
| TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306 |
| Check Table Before Drop: false                                    |
+-------------------------------------------------------------------+

```

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close #16224 

Problem Summary:
In some cases, the concurrency information of some executors will be recorded many times.

### What is changed and how it works?
Clear the concurrency information before we set the concurrency information.

### Check List 

Tests 

- Unit test
- Integration test

### Release note
- executor: Remove unnecessary information in explain analyze output

### CODE REVIEW

File: util/execdetails/execdetails.go [hunk 1]
```diff
@@ -457,6 +457,14 @@ func (e *RuntimeStats) SetRowNum(rowNum int64) {
 	atomic.StoreInt64(&e.rows, rowNum)
 }
 
+// ClearConcurrencyInfo clears the concurrency information.
+// We must call clearConcurrencyInfo first when we need to call SetConcurrencyInfo.
```
Comment (crazycs520) [inline util/execdetails/execdetails.go:461]: If so, how about call this function in `SetConcurrencyInfo` and remove in other place?

Comment (Reminiscent) [reply util/execdetails/execdetails.go:461]: In [`Hash Agg`](https://github.com/pingcap/tidb/blob/d62fcd408e19b24e6d5704d1f66fc652ce386ed5/executor/aggregate.go#L246-L248 ) , we have two concurrency information need to set. So I think it's better to keep two functions. Or we can change `SetConcurrencyInfo` to get more concurrency information and set them all at once. What's your opinion?  @crazycs520

Comment (zz-jason) [reply util/execdetails/execdetails.go:461]: > Or we can change SetConcurrencyInfo to get more concurrency information and set them all at once.

I prefer this way. The current implementation requires each executor to clear their concurrency information after execution, which is hardly to achieve by different contributors.

**Comment by Reminiscent** [comment]
Update. PTAL @zz-jason @crazycs520

File: util/execdetails/execdetails.go [hunk 2]
```diff
@@ -466,12 +471,16 @@ func (e *RuntimeStats) SetRowNum(rowNum int64) {
 	atomic.StoreInt64(&e.rows, rowNum)
 }
 
-// SetConcurrencyInfo sets the concurrency information.
+// SetConcurrencyInfo sets the concurrency informations.
+// We must clear the concurrencyInfo first when we call the SetConcurrencyInfo.
 // When the num <= 0, it means the exector operator is not executed parallel.
-func (e *RuntimeStats) SetConcurrencyInfo(name string, num int) {
+func (e *RuntimeStats) SetConcurrencyInfo(infos []*ConcurrencyInfo) {
```
Comment (zz-jason) [inline util/execdetails/execdetails.go:477]: how about:
```suggestion
func (e *RuntimeStats) SetConcurrencyInfo(infos ...*ConcurrencyInfo) {
```

Comment (Reminiscent) [reply util/execdetails/execdetails.go]: Update. @zz-jason PTAL

**Comment by zz-jason** [comment]
@Reminiscent do we need to cherry-pick this bugfix to the release branches?

**Pull request merged**